### ISSUE TITLE
DB2: Add support for more labeled durations e.g. `MONTH`, `YEAR`

### ISSUE THREAD

**Issue by keraion** [sqlfluff/sqlfluff#5610]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Sqlfluff fails to parse with two things:
- Labeled durations that include any expression or column reference
- Labeled durations besides `DAY` or `DAYS`

### Expected Behaviour

Parsing should be successful.

### Observed Behaviour

Given the following statement:
```sql
SELECT *
FROM TABX
WHERE MY_DT BETWEEN CURRENT_DATE - (DAYS(CURRENT_DATE) -1) DAYS - 180 DAYS
AND CURRENT_DATE;
```

```sh
sqlfluff parse --dialect db2 test.sql

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    wildcard_expression:
[L:  1, P:  8]      |                        wildcard_identifier:
[L:  1, P:  8]      |                            star:                             '*'
[L:  1, P:  9]      |                [META] dedent:
[L:  1, P:  9]      |            newline:                                          '\n'
[L:  2, P:  1]      |            from_clause:
[L:  2, P:  1]      |                keyword:                                      'FROM'
[L:  2, P:  5]      |                whitespace:                                   ' '
[L:  2, P:  6]      |                from_expression:
[L:  2, P:  6]      |                    [META] indent:
[L:  2, P:  6]      |                    from_expression_element:
[L:  2, P:  6]      |                        table_expression:
[L:  2, P:  6]      |                            table_reference:
[L:  2, P:  6]      |                                naked_identifier:             'TABX'
[L:  2, P: 10]      |                    [META] dedent:
[L:  2, P: 10]      |            newline:                                          '\n'
[L:  3, P:  1]      |            where_clause:
[L:  3, P:  1]      |                keyword:                                      'WHERE'
[L:  3, P:  6]      |                [META] (implicit) indent:
[L:  3, P:  6]      |                whitespace:                                   ' '
[L:  3, P:  7]      |                expression:
[L:  3, P:  7]      |                    column_reference:
[L:  3, P:  7]      |                        naked_identifier:                     'MY_DT'
[L:  3, P: 12]      |                [META] dedent:
[L:  3, P: 12]      |            whitespace:                                       ' '
[L:  3, P: 13]      |            unparsable:                                       !! Expected: 'Nothing here.'
[L:  3, P: 13]      |                word:                                         'BETWEEN'
[L:  3, P: 20]      |                whitespace:                                   ' '
[L:  3, P: 21]      |                word:                                         'CURRENT_DATE'
[L:  3, P: 33]      |                whitespace:                                   ' '
[L:  3, P: 34]      |                minus:                                        '-'
[L:  3, P: 35]      |                whitespace:                                   ' '
[L:  3, P: 36]      |                start_bracket:                                '('
[L:  3, P: 37]      |                word:                                         'DAYS'
[L:  3, P: 41]      |                start_bracket:                                '('
[L:  3, P: 42]      |                word:                                         'CURRENT_DATE'
[L:  3, P: 54]      |                end_bracket:                                  ')'
[L:  3, P: 55]      |                whitespace:                                   ' '
[L:  3, P: 56]      |                minus:                                        '-'
[L:  3, P: 57]      |                numeric_literal:                              '1'
[L:  3, P: 58]      |                end_bracket:                                  ')'
[L:  3, P: 59]      |                whitespace:                                   ' '
[L:  3, P: 60]      |                word:                                         'DAYS'
[L:  3, P: 64]      |                whitespace:                                   ' '
[L:  3, P: 65]      |                minus:                                        '-'
[L:  3, P: 66]      |                whitespace:                                   ' '
[L:  3, P: 67]      |                numeric_literal:                              '180'
[L:  3, P: 70]      |                whitespace:                                   ' '
[L:  3, P: 71]      |                word:                                         'DAYS'
[L:  3, P: 75]      |                newline:                                      '\n'
[L:  4, P:  1]      |                word:                                         'AND'
[L:  4, P:  4]      |                whitespace:                                   ' '
[L:  4, P:  5]      |                word:                                         'CURRENT_DATE'
[L:  4, P: 17]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    statement_terminator:                                     ';'
[L:  5, P:  2]      |    newline:                                                  '\n'
[L:  6, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   3 | P:  13 |  PRS | Line 3, Position 13: Found unparsable section: 'BETWEEN
                       | CURRENT_DATE - (DAYS(CURRENT_DAT...'
WARNING: Parsing errors found and dialect is set to 'db2'. Have you configured your dialect correctly?
```

Parsing also fails with duration types other than `DAY(S)`:
```sh
echo 'VALUES NOW + 1 MICROSECOND;' | sqlfluff parse --dialect db2 -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        values_clause:
[L:  1, P:  1]      |            keyword:                                          'VALUES'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            expression:
[L:  1, P:  8]      |                column_reference:
[L:  1, P:  8]      |                    naked_identifier:                         'NOW'
[L:  1, P: 11]      |                whitespace:                                   ' '
[L:  1, P: 12]      |                binary_operator:                              '+'
[L:  1, P: 13]      |                whitespace:                                   ' '
[L:  1, P: 14]      |                numeric_literal:                              '1'
[L:  1, P: 15]      |    whitespace:                                               ' '
[L:  1, P: 16]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  1, P: 16]      |        word:                                                 'MICROSECOND'
[L:  1, P: 27]      |        semicolon:                                            ';'
[L:  1, P: 28]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:  16 |  PRS | Line 1, Position 16: Found unparsable section: 'MICROSECOND;'
WARNING: Parsing errors found and dialect is set to 'db2'. Have you configured your dialect correctly?
```

### How to reproduce

See observed behavior examples.

### Dialect

DB2

### Version

sqlfluff, version 3.0.0a5

### Configuration

defaults

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
